istioctl/pkg/install/pre-check.go

```diff
@@ -300,3 +319,186 @@ func (c *preCheckClient) checkMutatingWebhook() error {
+	var (
+		kubeConfigFlags = &genericclioptions.ConfigFlags{
+			Context:    strPtr(""),
+			Namespace:  strPtr(""),
+			KubeConfig: strPtr(""),
+		}
+
+		filenames     = []string{}
+		fileNameFlags = &genericclioptions.FileNameFlags{
+			Filenames: &filenames,
+			Recursive: boolPtr(false),
+			Usage:     "Istio YAML installation file.",
+		}
+		istioNamespace string
+		opts           clioptions.ControlPlaneOptions
+	)
+	precheckCmd := &cobra.Command{
+		Use:   "precheck [-f <deployment or istio operator file>]",
+		Short: "Checks Istio cluster compatibility",
+		Long: `
+		precheck inspects a Kubernetes cluster for Istio install requirements.
+`,
+		Example: `
+		# Verify that Istio can be installed
+		istioctl experimental precheck
+
+		# Verify the deployment matches a custom Istio deployment configuration
+		istioctl x precheck --set profile=demo
+
+		# Verify the deployment matches the Istio Operator deployment definition
+		istioctl x precheck -f iop.yaml
+`,
+		Args: cobra.ExactArgs(0),
+		RunE: func(c *cobra.Command, args []string) error {
+			targetNamespace := istioNamespace
+			targetRevision := opts.Revision
+			specific := c.Flags().Changed("istioNamespace") // is user asking about a specific Istio System ns or revision
+
+			// Check if we can install the IOP specified with -f
+			if len(fileNameFlags.ToOptions().Filenames) > 0 {
+				iop, err := getIOPFromFile(kubeConfigFlags, fileNameFlags.ToOptions())
+				if err != nil {
+					return err
+				}
+				// Currently we don't look at specific IOP options, just the namespace and Revision
+				targetNamespace = iop.GetNamespace()
+				targetRevision = iop.Spec.Revision
+				specific = true
+			}
+
+			cli, err := clientFactory(kubeConfigFlags)
+			if err != nil {
+				return err
+			}
+
+			installs, err := cli.getIstioInstalls()
+			if err == nil && len(installs) > 0 {
+				matched := false
+				for _, install := range installs {
+					if !specific || targetNamespace == install.namespace && targetRevision == install.revision {
+						c.Printf("Istio Revision %q already installed in namespace %q\n", install.revision, install.namespace)
+					}
+					if targetNamespace == install.namespace && targetRevision == install.revision {
+						matched = true
+					}
+				}
+				// The user has Istio, but wants to install a new revision
+				if !matched {
+					return installPreCheck(targetNamespace, kubeConfigFlags, c.OutOrStdout())
+				}
+				return nil
+			}
+
+			// No IstioOperator was found.  In 1.6.0 we fall back to checking for Istio namespace
+			nsExists, err := namespaceExists(targetNamespace, kubeConfigFlags)
+			if err != nil {
+				return err
+			}
+			if !nsExists {
+				return installPreCheck(targetNamespace, kubeConfigFlags, c.OutOrStdout())
+			}
+
+			// The Istio namespace does exist, but it wasn't installed by 1.6.0+ because no
+			// IstioOperator is there.
+			c.Printf("Istio already installed in namespace %q\n.  Skipping pre-check.", targetNamespace)
```

Comment:
Move `\n` in the end:
c.Printf("Istio already installed in namespace %q.  Skipping pre-check . \n ", targetNamespace)
Now it is a little strange:
```
Istio already installed in namespace "istio-system"
.  Skipping pre-check.Use 'istioctl upgrade' to upgrade or 'istioctl install --set revision=<revision>' to install another control plane.
```

Reply:
fixed